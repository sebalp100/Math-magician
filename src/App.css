@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600&display=swap');

* {
  margin: 0;
  padding: 0;
}

body {
  background-color: rgb(44, 44, 44);
  color: azure;
  font-family: 'Poppins', sans-serif;
}

.calc-wrapper {
  display: flex;
  justify-content: center;
  height: 100vh;
  align-items: center;
}

.main-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

table {
  display: flex;
  border-collapse: collapse;
}

::-webkit-scrollbar {
  display: none;
}

.output {
  width: 100%;
  background-color: darkgray;
  border: none;
  text-align: right;
  outline-style: none;
  height: 40px;
  font-size: 20px;
  color: white;
  max-width: 320px;
  overflow-x: scroll;
  -ms-overflow-style: none;
  scrollbar-width: none;
}

.input {
  padding-right: 5px;
  background-color: darkgray;
}

button {
  width: 80px;
  height: 40px;
  border-radius: 0;
  border: 0.5px solid gray;
  cursor: pointer;
}

.zero {
  width: 50%;
}

.orange {
  background-color: rgb(241, 142, 106);
}

nav {
  display: flex;
  justify-content: space-between;
  padding-left: 4%;
  padding-top: 20px;
}

h1 {
  font-size: 27px;
}

.link {
  display: flex;
  list-style: none;
  gap: 20px;
  padding-right: 4%;
}

.homeContainer {
  display: flex;
  flex-direction: column;
  padding: 4%;
  gap: 40px;
}

.calc-container,
.quoteContainer {
  display: flex;
  flex-direction: column;
  padding: 4%;
  align-items: center;
}

.quoteContainer {
  padding-top: 50px;
  font-size: 20px;
}

@media only screen and (min-width: 768px) {
  .link {
    gap: 100px;
  }

  .calc-container {
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
  }

  .box {
    display: flex;
    justify-content: center;
    width: 50%;
    height: 50vh;
    border: 2px solid white;
    background-color: dimgray;
  }
}
