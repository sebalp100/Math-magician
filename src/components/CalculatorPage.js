import React from 'react';
import Calculator from './Calculator';

const CalculatorPage = () => (
  <div className="calc-container">
    <h3>LetÂ´s do some math!</h3>
    <div className="box">
      <Calculator />
    </div>
  </div>
);

export default CalculatorPage;
