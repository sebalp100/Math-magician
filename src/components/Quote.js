import React from 'react';

const Quote = () => (
  <div className="homeContainer">
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eu feugiat ex,
      et fermentum ipsum. Vestibulum ultrices dapibus enim egestas dictum. Suspendisse
      facilisis, purus et finibus tincidunt, nulla nulla eleifend mi, sit amet ullamcorper
      tellus ipsum in mi. Fusce imperdiet, augue nec iaculis congue, est mi accumsan elit,
      eget hendrerit risus leo a mauris. Donec id pretium eros, nec gravida metus.
    </p>
  </div>
);

export default Quote;
