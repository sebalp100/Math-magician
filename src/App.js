import './App.css';
import React from 'react';
import Calculator from './components/Calculator';

const App = () => (
  <div className="calc-wrapper">
    <Calculator />
  </div>
);

export default App;
