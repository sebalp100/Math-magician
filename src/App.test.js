import * as React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Quote from './components/Quote';
import CalculatorPage from './components/CalculatorPage';

describe('App', () => {
  it('renders home component', () => {
    render(<App />);

    expect(screen.getByText(/Home/)).toBeInTheDocument();
    screen.debug();
  });
});

describe('Quote', () => {
  it('renders Qoute component', () => {
    render(<Quote />);
  });
});

describe('Calculator', () => {
  it('renders CalculatorPage component', () => {
    render(<CalculatorPage />);
  });
});

test('First Snapshot test', () => {
  const component = renderer
    .create(<App />)
    .toJSON();

  expect(component).toMatchSnapshot();
});
