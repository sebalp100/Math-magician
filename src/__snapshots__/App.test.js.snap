// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Snapshot test 1`] = `
Array [
  <nav>
    <h1>
      Math Magicians
    </h1>
    <div
      className="link"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "azure",
              "textDecoration": "none",
            }
          }
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
          style={
            Object {
              "color": "azure",
              "textDecoration": "none",
            }
          }
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
          style={
            Object {
              "color": "azure",
              "textDecoration": "none",
            }
          }
        >
          Quote
        </a>
      </li>
    </div>
  </nav>,
  <div
    className="homeContainer"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean eu feugiat ex, et fermentum ipsum. Vestibulum ultrices dapibus enim egestas dictum. Suspendisse facilisis, purus et finibus tincidunt, nulla nulla eleifend mi, sit amet ullamcorper tellus ipsum in mi. Fusce imperdiet, augue nec iaculis congue, est mi accumsan elit, eget hendrerit risus leo a mauris.
      <br />
      <br />
      Donec id pretium eros, nec gravida metus. Vestibulum commodo turpis vel porta efficitur. Sed commodo, metus ut interdum euismod, nulla massa tempor mi, et suscipit mi quam vitae eros. Morbi nec lorem eget tortor tincidunt convallis. In tristique eleifend mauris ac maximus. Sed ut ultricies sapien, vitae fringilla eros.
    </p>
  </div>,
]
`;
